vcd/resource_vcd_vapp_firewall_rules.go

```diff
@@ -0,0 +1,390 @@
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"log"
+	"strings"
+)
+
+func resourceVcdVappFirewallRules() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdVapFirewallRulesCreate,
+		Delete: resourceVAppFirewallRulesDelete,
+		Read:   resourceVappFirewallRulesRead,
+		Update: resourceVcdVapFirewallRulesUpdate,
+		Importer: &schema.ResourceImporter{
+			State: vappFirewallRuleImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"vapp_id": &schema.Schema{
```

Comment:
It's good to see we're moving on to using IDs. However, why this decision has been made particularly in this resource? Is there some technical reason specific to the vApp FW rules VCD API or just us expediting switch to IDs?

Reply:
We are using IDs already in several resources. Look at `vcd_nsxv_firewall_rule` for example.
What we have designed before starting the implementation of data sources is name-oriented. With data sources available, referencing by ID becomes easy

Reply:
Guides may be missed. Looking again, actually it's the data source doc pages that would benefit the most from a few additional `output` lines in the top examples which show how to get the ID. @vbauzysvmware could you try whether they return the ID we expect and add such snippets to the EG, vApp and VM data source pages as part of this PR? I mean near to the current samples like:
```
output "name" {
  value = data.vcd_vapp.test-tf.name
}
```

Reply:
added

Reply:
Thanks!!